services/datamanager/datasync/sync.go

```diff
@@ -34,6 +34,9 @@ var (
 // FailedDir is a subdirectory of the capture directory that holds any files that could not be synced.
 const FailedDir = "failed"
 
+// maxParallelSyncRoutines is the maximum number of sync goroutines that can be running at once.
```

Comment:
Can make this configurable within the data manager service, but keeping simple for v0 to avoid bloating the config.

Reply:
I'm actually thinking about going lower. Do you have the bandwidth to repeat the test with a change? I'm curious what happens if (a typical) deployment hits the ulimit for open file handles. It's common for the limit to be set to 1024. Keeping in mind (IIRC) socket connections are also file handles. So something like:
```
ulimit -n 200
<run test>
```

If you want to observe the ulimit was set, you can do `ulimit -n`. Or `ulimit -a` which gives you a bunch of nice names:
```
dan.gottlieb@instance-template-1:~/rdk/vendor$ ulimit -a
real-time non-blocking time  (microseconds, -R) unlimited
core file size              (blocks, -c) 0
data seg size               (kbytes, -d) unlimited
scheduling priority                 (-e) 0
file size                   (blocks, -f) unlimited
pending signals                     (-i) 31431
max locked memory           (kbytes, -l) 1016871
max memory size             (kbytes, -m) unlimited
open files                          (-n) 1024
pipe size                (512 bytes, -p) 8
POSIX message queues         (bytes, -q) 819200
real-time priority                  (-r) 0
stack size                  (kbytes, -s) 8192
cpu time                   (seconds, -t) unlimited
max user processes                  (-u) 31431
virtual memory              (kbytes, -v) unlimited
file locks                          (-x) unlimited
```

Then just looking out to see if the program crashes or gets weird errors when trying to connect to it from a control page.

If rerunning the experiment is problematic, I'm happy to do it with some hand holding to get started.

Reply:
Thanks laying out the test. Tested that this works, and verified via both `ulimit -n` and `ulimit -a` that the open file limit was set to 200. No crashing, syncing continues, control page continues to work.